proto/sources.proto

```diff
@@ -270,3 +271,14 @@ message SlackRealtime {
     credentials.SlackTokens tokens = 1;
   }
 }
+
+message Sharepoint {
+  string endpoint = 1 [(validate.rules).string.uri_ref = true];
+  oneof credential {
+    string token = 2;
+    credentials.ClientCredentials authenticated = 3;
+    credentials.Oauth2 oauth = 4;
```

Comment:
remove this, we don't need to provide a client secret or ID, just the refresh token.